File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):

    count = [0] * (number + 1)


    for i in range(1, number + 1):
        for j in range(1, (number // i) + 1):
            count[i * j] += 1


    for i in range(len(count)):
        if count[i] > limit:
            count[i] = power
    return sum(count)

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.Arrays;

class Solution {

    public int solution(int number, int limit, int power) {

        int[] count = new int[number + 1];

        for (int i = 1; i <= number; i++) {
            for (int j = 1; j <= number / i; j++) {
                count[i * j]++;
            }
        }

        return Arrays.stream(count).map(i -> i > limit ? power : i).sum();
        
    }
}